ca_tools/scripts/pos_controller_pid.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+import rospy
+import math
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Twist
+from tf.transformations import euler_from_quaternion
+from geometry_msgs.msg import Pose2D
+
+
+class DrawSquare:
+
+    # Time variables
+    Now=0.0
+    TimeChange=0.0
+    LastTime=0.0
+
+    # Angle variables
+
+    CurrentAngle=0.0
+    SetpointAngle=0.0
+    InputAngle=0.0
+    ErrorAngle=0.0
+    ErrSumAngle=0.0
+    MaxOutputAngle=0.0
+    MinOutputAngle =0.0
+
+    # Lenght variables
+
+    CurrentLength=0.0
+    SetpointLength=Pose2D()
+    InputLength=0.0
+    ErrorLength=0.0
+    ErrSumLength=0.0
+    MaxOutputLength=0.0
+    MinOutputLength=0.0
+
+    # PID variables
+    Kp=0
+    Ki=0
+    Kd=0
+
+    # Auxiliar Variables
+    CurrAux = Pose2D()
+    IniAux=Pose2D()
+    FinAux=Pose2D()
+    ErrorLengthAux=Pose2D()
+
+    #State Variables
+    #Set Variable is TRUE if we need to set new goals
+
+    SetVariable=True
+    State="FORWARD"
+
+    #Twist Variable
+    Vel = Twist()
+
+    #Odometry Flag
+    OdomFlag=0
+
+    Output_angle = 0.0
+    Output_length = 0.0
+
+
+    def __init__(self):
+
+        # PID variables
+        self.Kp=5
+        self.Ki=2
+        self.Kd=0
+
+        #Range of output
+
+        self.MaxOutputAngle = 0.4
+        self.MinOutputAngle = 0.4
+
+        self.MaxOutputLength = 0.4
+        self.MinOutputLength = 0.4
+
+        #self.Setpoint_angle = self.Input_angle
+        self.SetVariable = True
+        self.State="FORWARD"
+
+    def callback(self,data):
+
+        q = (data.pose.pose.orientation.x,
+            data.pose.pose.orientation.y,
+            data.pose.pose.orientation.z,
+            data.pose.pose.orientation.w)
+
+        _, _, self.CurrentAngle = euler_from_quaternion(q)
+
+        self.CurrAux.x=data.pose.pose.position.x
+        self.CurrAux.y=data.pose.pose.position.y
+
+        self.OdomFlag+=1
```

Comment:
Two things regarding this flag:

1. If the value will be between 0 and 1, why don't you use a boolean?
2. What happens if you remove the flag and the update loop updates your math every time? Is there an issue with that?